app/app.rb

```diff
@@ -0,0 +1,78 @@
+ENV['RACK_ENV'] ||= 'development'
+
+require 'sinatra/base'
+require 'sinatra/flash'
+require_relative 'dm_setup'
+
+class Chitter < Sinatra::Base
+  enable :sessions
+  set :session_secret, 'VERY secure secret DO NOT STEAL'
+  register Sinatra::Flash
+  use Rack::MethodOverride
+
+  get '/' do
+    erb :index
+  end
+
+  get '/peeps' do
+    @peeps = Peep.all
+    erb :'peeps/peeps'
+  end
+
+  get '/peeps/new' do
+    erb :'peeps/new'
+  end
+
+  post '/peeps/new' do
+    if current_user
+      Peep.create(message: params[:message], user_id: session[:user_id])
+      redirect to '/peeps'
+    else
+      flash.now[:errors] = "You must sign in or register to post a peep"
+      erb :'/peeps/new'
+    end
+  end
+
+  get '/users/new' do
+    erb :'/users/new'
+  end
+
+  post '/users/new' do
+    @user = User.new(username: params[:username], email: params[:email],
+                    password: params[:password],
+                    password_confirmation: params[:password_confirmation])
+    if @user.save
+      session[:user_id] = @user.id
```

Comment:
Can this line be made more declarative of what it does? At the moment, it seems to reveal its implementation, rather than its intention.